### PR TITLE
Update cd-hit tool v.4.8.1

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/cdhit/cd_hit.xml [hunk 1]
```diff
@@ -140,6 +141,8 @@ $advanced.inram
...
             <param name="inram" argument="-B" type="boolean" truevalue="-B 0" falsevalue="-B 1" checked="true" label="Sequences are stored in RAM" help="If false: sequence are stored on hard drive - use for huge data sets"/>
             <param name="accurate" argument="-g" type="boolean" truevalue="-g 1" falsevalue="-g 0" checked="false" label="Accurate but slow mode" help="By cd-hit's default algorithm, a sequence is clustered to the first cluster that meet the threshold (fast cluster). If set to true, the program will cluster it into the most similar cluster that meet the threshold (accurate but slow mode)"/>
+            <param name="sort_cluster" argument="-sc" type="boolean" truevalue="-sc 1" falsevalue="-sc 0" label="Sort clusters by size" help="When disabled, clusters are sorted by decreasing length; if enabled, clusters are sorted by decreasing size" />
+            <param name="sort_fasta" argument="-sf" type="boolean" truevalue="-sf 1" falsevalue="-sf 0" label="Sort fasta/fastq by cluster size" help="When enabled, output sequences are sorted by decreasing cluster size" />
```
Comment (nsoranzo) [inline tools/cdhit/cd_hit.xml:145]:
```suggestion
            <param name="sort_fasta" argument="-sf" type="boolean" truevalue="-sf 1" falsevalue="-sf 0" label="Sort FASTA/FASTQ by cluster size" help="When enabled, output sequences are sorted by decreasing cluster size" />
```

File: tools/cdhit/test-data/sorted_fasta.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
+>MISEQ-M4070_81_000000000-B4JFH_1_1119_8913_6810_TACCGTAG
```
Comment (nsoranzo) [inline tools/cdhit/test-data/sorted_fasta.txt:1]: This file doesn't seem to be sorted, it's the same as `tools/cdhit/test-data/est-2d.txt`

File: tools/cdhit/test-data/sorted_cluster.txt.clstr [hunk 3]
```diff
@@ -0,0 +1,868 @@
+>Cluster 0
```
Comment (nsoranzo) [inline tools/cdhit/test-data/sorted_cluster.txt.clstr:1]: This file is the same as `tools/cdhit/test-data/est-2d.txt.clstr`, is this correct?

Comment (gallardoalba) [reply tools/cdhit/test-data/sorted_cluster.txt.clstr:1]: I'll check that.

File: tools/cdhit/cd_hit.xml [hunk 4]
```diff
@@ -279,8 +318,5 @@ For cd-hit-2d and cd-hit-est-2d:
 
 2. The second output is a text file that lists similar sequences between db1 & db2
     ]]></help>
-    <citations>
-        <citation type="doi">10.1093/bioinformatics/btl158</citation>
-        <citation type="doi">10.1093/bioinformatics/bts565</citation>
-    </citations>
+    <expand macro='citations'/>
```
Comment (nsoranzo) [inline tools/cdhit/cd_hit.xml:321]:
```suggestion
    <expand macro="citations" />
```

File: tools/cdhit/macros.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<macros>
+    <token name="@TOOL_VERSION@">4.8.1</token>
+    <token name="@VERSION_SUFFIX@">0</token>
+    <xml name="xrefs">
+      <xrefs>
+        <xref type="bio.tools">cd-hit</xref>
+      </xrefs>
```
Comment (nsoranzo) [inline tools/cdhit/macros.xml:7]:
```suggestion
        <xrefs>
            <xref type="bio.tools">cd-hit</xref>
        </xrefs>
```

File: tools/cdhit/macros.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    </xml>
+    <xml name="requirements">
+      <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">cd-hit</requirement>
+      </requirements>
```
Comment (nsoranzo) [inline tools/cdhit/macros.xml:12]:
```suggestion
        <requirements>
            <requirement type="package" version="@TOOL_VERSION@">cd-hit</requirement>
        </requirements>
```

File: tools/cdhit/macros.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    </xml>
+    <xml name="citations">
+      <citations>
+        <citation type="doi">10.1093/bioinformatics/btl158</citation>
+        <citation type="doi">10.1093/bioinformatics/bts565</citation>
+      </citations>
```
Comment (nsoranzo) [inline tools/cdhit/macros.xml:18]:
```suggestion
        <citations>
            <citation type="doi">10.1093/bioinformatics/btl158</citation>
            <citation type="doi">10.1093/bioinformatics/bts565</citation>
        </citations>
```

File: tools/cdhit/test-data/sorted_cluster.txt.clstr (hunk 3 above)
Comment (nsoranzo) [reply tools/cdhit/test-data/sorted_cluster.txt.clstr]: Maybe https://github.com/weizhongli/cdhit/issues/116 is related? Not sure it's worth adding the options if we are not sure they work.

**Review by bgruening — APPROVED** [review]

**Pull request merged**